packages/chat-component/src/main.ts

```diff
@@ -105,8 +105,30 @@ export class ChatComponent extends LitElement {
 
   static override styles = [mainStyle];
 
-  // Send the question to the Open AI API and render the answer in the chat
+  override connectedCallback(): void {
+    super.connectedCallback();
 
+    // add event listener to know when to scroll chat list footer into view
+    this.addEventListener('must-scroll', () => {
+      this.handleOnMustScroll();
+    });
+  }
+  // debounce dispatching must-scroll event
+  debounceDispatchMustScrollEvent(): void {
+    let timeout: any = 0;
+    clearTimeout(timeout);
+    timeout = setTimeout(() => {
+      this.dispatchEvent(mustScrollEvent);
+    }, 500);
+  }
+  // handle must scroll event
+  handleOnMustScroll(): void {
+    const footer = this.shadowRoot?.querySelector('#chat-list-footer') as HTMLElement;
```

Comment:
is there an idiomatic `lit`-way to do this?

Reply:
Looks we can use `@query()` https://lit.dev/docs/components/shadow-dom/#query

Reply:
Fixed!